common/OpTestLpar.py

```diff
@@ -477,6 +482,322 @@ def lpar_clone_linux_source(self, i_dir):
+    #
+    def lpar_check_pkg_for_utility(self, i_oslevel, i_cmd):
+        if 'Ubuntu' in i_oslevel:
+            l_res = self._ssh_execute("dpkg -S `which %s`" % i_cmd)
+            return l_res
+        else:
+            l_cmd = "rpm -qf `which %s`" % i_cmd
+            l_res = self._ssh_execute(l_cmd)
+            l_pkg = l_res.replace("\r\n", "")
+            print l_pkg
+            return l_pkg
+
+    ##
+    # @brief This function loads ibmpowernv driver only on powernv platform
+    #        and also this function works only in root user mode
+    #
+    # @param i_oslevel @type string: OS level
+    #
+    # @return BMC_CONST.FW_SUCCESS or raise OpTestError
+    #
+    def lpar_load_ibmpowernv(self, i_oslevel):
+        if "PowerKVM" not in i_oslevel:
+            l_rc = self._ssh_execute("modprobe ibmpowernv; echo $?")
+            l_rc = l_rc.replace("\r\n", "")
+            if int(l_rc) == 0:
+                cmd = "lsmod | grep -i ibmpowernv"
+                response = self._ssh_execute(cmd)
+                if "ibmpowernv" not in response:
+                    l_msg = "ibmpowernv module is not loaded, exiting"
+                    raise OpTestError(l_msg)
+                else:
+                    print "ibmpowernv module is loaded"
+                print cmd
+                print response
+                return BMC_CONST.FW_SUCCESS
+            else:
+                l_msg = "modprobe failed while loading ibmpowernv,exiting..."
+                print l_msg
+                raise OpTestError(l_msg)
+        else:
+            return BMC_CONST.FW_SUCCESS
+
+    ##
+    # @brief This function restarts the lm_sensors service on lpar using systemctl utility
+    #        systemctl utility is not present in ubuntu, This function will work in remaining all
+    #        other OS'es i.e redhat, sles and PowerKVM
+    #
+    # @param i_oslevel @type string: OS level
+    #
+    # @return BMC_CONST.FW_SUCCESS or raise OpTestError
+    #
+    def lpar_start_lm_sensor_svc(self, i_oslevel):
+        if 'Ubuntu' in i_oslevel:
+            pass
+        else:
+            try:
+                # Start the lm_sensors service
+                cmd = "/bin/systemctl stop  lm_sensors.service"
+                self.lpar_run_command(cmd)
+                cmd = "/bin/systemctl start  lm_sensors.service"
+                self.lpar_run_command(cmd)
+                cmd = "/bin/systemctl status  lm_sensors.service"
+                res = self.lpar_run_command(cmd)
+                return BMC_CONST.FW_SUCCESS
+            except:
+                l_msg = "loading lm_sensors service failed"
+                print l_msg
+                raise OpTestError(l_msg)
+
+    ##
+    # @brief It will clone latest linux git repository in i_dir directory
+    #
+    # @param i_dir @type string: directory where linux source will be cloned
+    #
+    # @return BMC_CONST.FW_SUCCESS or raise OpTestError
+    #
+    def lpar_clone_linux_source(self, i_dir):
```

Comment:
Need to figure out why this is showing up as a diff in this review.  Maybe you need a rebase on master?